mirrord/agent/src/cgroup.rs

```diff
@@ -0,0 +1,256 @@
+//! Logic for pausing using cgroup directly
+//! Pause requires privileged - assumes ephemeral (hardcoded pid 1)
+use std::{
+    fs::OpenOptions,
+    io::{BufRead, BufReader, Write},
+    path::{Path, PathBuf},
+};
+
+use enum_dispatch::enum_dispatch;
+use nix::mount::{mount, MsFlags};
+use tracing::{trace, warn};
+
+use crate::{
+    error::{AgentError, Result},
+    namespace::{set_namespace, NamespaceType},
+};
+
+const CGROUP_MOUNT_PATH: &str = "/mirrord_cgroup";
+const CGROUP_SUBGROUP_MOUNT_PATH: &str = "/mirrord_cgroup/subgroup";
+const CGROUPV2_PROCS_FILE: &str = "cgroup.procs";
+
+/// Trait for objects that can be paused
+#[enum_dispatch]
```

Comment:
Why use `enum_dispatch`? Are we going to implement this for more than 2 cgroup types?

Can't we just have a normal enum 

```rust
enum Cgroup {
    V1(CgroupV1),
    V1(CgroupV2),
}
```

Implement the differences in the inner structs, e.g.:

```rust
impl CgroupV1 {
    fn new(...) -> Result<Self> { ... }
}
```

and the parts that are the same in the enum impl:

```rust
impl Cgroup {
    fn freeze(&mut self) -> Result<()> {
        match self {
            V1(v1) => v1.freeze(),
            V2(v2) => v2.freeze(),
        }
    }
}
```

and if you really want, you can have a trait for the inner structs to implement:

```rust
trait Freezer {
    fn freeze(&self);
    fn unfreeze(&self);
}

impl Freezer for CgroupV1 { ... }
impl Freezer for CgroupV2 { ... }
```

Then we can use tokio IO stuff and have the file operations be `async`.